### PR TITLE
refactor(framework): move src files

### PR DESCRIPTION
**Pull request by khaziwallis**
JIRA: https://jira.rax.io/browse/EO-1082

### LGTMs
- [x] Dev LGTM
- [x] Dev LGTM
- [x] Design LGTM
- [x] QE LGTM

Before
=======

Components -> rxSelectFilter
-----------------------------------

before-1

API Documentation -> rxSelectFilter -> rxSelectFilter
-------------------------------------------------------------

before-2

Tables link didn't display for rxSelectFilter
------------------------------------------------

before-3

Tables link didn't display for dropdown
------------------------------------------------
before-4

Tables link didn't display for form
------------------------------------------------
before-5

Tables link didn't display for rxSearchBox
------------------------------------------------
before-6

Tables link didn't display for input
------------------------------------------------
before-7

Tables link didn't display for select
------------------------------------------------
before-8

After
=======
Elements -> Tables -> rxSelectFilter
----------------------------------------
after-1

Elements -> Tables -> rxSelectFilter
----------------------------------------
screen shot 2016-07-22 at 1 48 14 pm

API Documentation -> elements -> rxSelectFilter
----------------------------------------
after-4

Tables link displayed for Tables
----------------------------------------
after-5

Tables link displayed for rxSelectFilter
----------------------------------------
after-6

Tables link displayed for dropdown
----------------------------------------
after-7

Tables link displayed for form
----------------------------------------
after-8

Table link displayed for search
----------------------------------------
after-9

Tables link displayed for input
----------------------------------------
after-10

Tables link displayed for select
----------------------------------------
after-11

### CODE REVIEW

**Comment by Droogans** [comment]
I'm seeing duplicated `"Tables"` entries under the Elements -> Tables section.

**Comment by Droogans** [comment]
I think this may be because the patterns used in [the Form section of Elements](https://github.com/rackerlabs/encore-ui/tree/f109c95af630bf99f52ccfda07e2bec9b1a0b69b/src/elements/Forms) is not being followed here.

**Comment by satyamkotakonda** [comment]
@Droogans  Do i have to display the "rxSelectFilter" information with in "Table" page.
http://rackerlabs.github.io/encore-ui/#/elements/Tables . Please conform.

**Comment by satyamkotakonda** [comment]
@Droogans  Can you please look into screen shot. Is it fine to display submenu ? or do i have to put the "rxSelectFilter" related information in "Table" page itself. (http://rackerlabs.github.io/encore-ui/#/elements/Tables )

unable to disply sub menu infromation

**Comment by Droogans** [comment]
@satyamkotakonda The related Jira story should have all the necessary information to complete this story.

However, I will say that if you transform the `Elements -> Tables` link into a submenu, the pre-existing content under the Tables link will be unreachable by users.

I would imagine that this content would be added as a section in the `Elements -> Tables` page, not in a submenu item.

**Comment by satyamkotakonda** [comment]
@Droogans Thanks very much for your suggestion.

**Comment by satyamkotakonda** [comment]
Before
=======

Components -> rxSelectFilter
-----------------------------------

before-1

API Documentation -> rxSelectFilter -> rxSelectFilter
-------------------------------------------------------------

before-2

Tables link didn't display for rxSelectFilter
------------------------------------------------

before-3

Tables link didn't display for dropdown
------------------------------------------------
before-4

Tables link didn't display for form
------------------------------------------------
before-5

Tables link didn't display for rxSearchBox
------------------------------------------------
before-6

Tables link didn't display for input
------------------------------------------------
before-7

Tables link didn't display for select
------------------------------------------------
before-8



After
=======
Elements -> Tables -> rxSelectFilter
----------------------------------------
after-1

Elements -> Tables -> rxSelectFilter
----------------------------------------
after-3

API Documentation -> elements -> rxSelectFilter
----------------------------------------
after-4

Tables link displayed for Tables
----------------------------------------
after-5

Tables link displayed for rxSelectFilter
----------------------------------------
after-6

Tables link displayed for dropdown
----------------------------------------
after-7

Tables link displayed for form
----------------------------------------
after-8

Table link displayed for search
----------------------------------------
after-9

Tables link displayed for input
----------------------------------------
after-10

Tables link displayed for select
----------------------------------------
after-11

**Comment by satyamkotakonda** [comment]
@Droogans As per your suggestions, updated changes and please review.
Thanks.

File: demo/scripts/demoApp.js [hunk 1]
```diff
@@ -262,7 +262,68 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
...
+                        },
+                        {
+                            linkText: 'Tags',
+                            href: '#/elements/Tags'
+                        },
+                        {
+                            linkText: 'Tooltips',
+                            href: '#/elements/Tooltips'
+                        }
+                    ]
```
Comment (100stacks) [inline demo/scripts/demoApp.js]: @satyamkotakonda: Please revert this change as it is not needed.

File: demo/scripts/demoApp.js [hunk 2]
```diff
@@ -262,7 +262,68 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
                 },
                 {
                     linkText: 'Elements',
-                    children: linksForModuleCategory('elements')
```
Comment (100stacks) [inline demo/scripts/demoApp.js]: @satyamkotakonda: Please revert this change as it is needed.

File: src/elements/Tables/Tables.meta.json [hunk 3]
```diff
@@ -24,6 +24,12 @@
...
         "sorting",
         "sort",
-        "status column"
+        "status column",
+        "rxSelectFilter",
+        "dropdown",
+        "form",
+        "search",
+        "input",
+        "select"
```
Comment (100stacks) [inline src/elements/Tables/Tables.meta.json]: Update `keywords` list to be in alphabetical order.

File: src/elements/Tables/docs/examples/rxSelectFilter.simple.html [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <!-- Sample HTML goes here as a live example of how the component can be used -->
-<div ng-controller="rxSelectFilterCtrl">
+<div ng-controller="rxSelectFilterSimpleExampleCtrl">
```
Comment (100stacks) [inline src/elements/Tables/docs/examples/rxSelectFilter.simple.html]: * Update controller to `selectFilterSimpleCtrl`
* Rename file to `selectFilter.simple.html`

File: src/elements/Tables/docs/examples/rxSelectFilter.simple.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
 angular.module('demoApp')
-.controller('rxSelectFilterCtrl', function ($scope, SelectFilter) {
+.controller('rxSelectFilterSimpleExampleCtrl', function ($scope, SelectFilter) {
```
Comment (100stacks) [inline src/elements/Tables/docs/examples/rxSelectFilter.simple.js]: * Update controller to `selectFilterSimpleCtrl`
* Rename file to `selectFilter.simple.js`

File: src/elements/Tables/docs/examples/rxSelectFilter.simple.html (hunk 4 above)
Comment (satyamkotakonda) [inline src/elements/Tables/docs/examples/rxSelectFilter.simple.html]: @100stacks Sorry, I am getting confused. Already I have done file rename  and this is path Tables/docs/examples/rxSelectFilter.simple.html.  Is it fine  or do i have to change any other places. Can you please guide me here.

File: src/elements/Tables/docs/examples/rxSelectFilter.simple.js (hunk 5 above)
Comment (satyamkotakonda) [inline src/elements/Tables/docs/examples/rxSelectFilter.simple.js]: @100stacks Sorry, I am getting confused. Already I have done file rename  and this is path "Tables/docs/examples/rxSelectFilter.simple.js".  Is it fine  or do i have to change any other places. Can you please guide me here.

Comment (100stacks) [inline src/elements/Tables/docs/examples/rxSelectFilter.simple.js]: @satyamkotakonda: Ok just the controller name is need to be updated.

File: src/elements/Tables/docs/examples/rxSelectFilter.simple.html (hunk 4 above)
Comment (100stacks) [inline src/elements/Tables/docs/examples/rxSelectFilter.simple.html]: @satyamkotakonda: Ok just the controller name is need to be updated.

**Comment by satyamkotakonda** [comment]
@100stacks  Thanks for your confirmation.

File: demo/scripts/demoApp.js [hunk 6]
```diff
@@ -165,6 +165,18 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
...
+            controller: 'showModuleController',
+            resolve: {
+                'module': function ($route, Modules) {
+                    return _.find(Modules, {
+                        'name': $route.current.params.subelement,
+                        category: 'elements'
+                    });
+                }
+            }
+        })
```
Comment (100stacks) [inline demo/scripts/demoApp.js]: @satyamkotakonda, this story does not require updates to `demo/scripts/demoApp.js`.

**Comment by satyamkotakonda** [comment]
@100stacks  Updated changes and please review.

File: src/elements/Tables/docs/tables.docs.html [hunk 7]
```diff
@@ -205,6 +205,11 @@
 <h3><rx-permalink>Nested Metadata</rx-permalink></h3>
 <rx-example name="table.nestedMetadata"></rx-example>
 
+<h2><rx-permalink>rxSelectFilter</rx-permalink></h2>
```
Comment (glynnis) [inline src/elements/Tables/docs/tables.docs.html]: Let's change this title to `Multi-select filters via rxSelectFilter`.

File: src/elements/Tables/docs/tables.docs.html [hunk 8]
```diff
@@ -205,6 +205,11 @@
...
+<p>
+  The rxSelectFilter component that provides a multi-select dropdown interface intended for table filtering.
+</p>
+<rx-example name="rxSelectFilter.simple"></rx-example>
```
Comment (glynnis) [inline src/elements/Tables/docs/tables.docs.html]: Can you move lines 108-212 up to line 189 so that this `rxSelectFilter` section appears with the other filters on the page?

File: src/elements/Tables/docs/tables.docs.html (hunk 7 above)
Comment (glynnis) [inline src/elements/Tables/docs/tables.docs.html]: Also change to `h3`, since we'll make this a subheader within the already existing "Filter" section on line 108.

**Comment by glynnis** [comment]
Hi @satyamkotakonda -- thanks for posting screen shots! I've added some design feedback where minor changes are needed. 👍

**Comment by 100stacks** [comment]
You will need to squash and rebase your commits once you incorporate @glynnis's feedback.

**Comment by 100stacks** [comment]
Also, please remember to squash this PR to two commits:
* first commit - just move files
* second commit - contains all edits/changes

**Comment by satyamkotakonda** [comment]
screen shot 2016-07-22 at 1 48 14 pm

**Comment by satyamkotakonda** [comment]
@glynnis   Updated changes and please review.

screen shot 2016-07-22 at 1 48 14 pm

**Comment by glynnis** [comment]
Thanks @satyamkotakonda ! This has a Design LGTM.

**Comment by 100stacks** [comment]
@Droogans, just curious why you removed the **`documentation`** `label` from the PRs.  Does it have a specific purpose meaning in the context of PRs?

**Comment by Droogans** [comment]
I was having trouble scanning the PR labels because many had 4+ labels on them.

Typically documentation-only PRs are marked "Documentation", and the "Testing" label should probably be deleted since almost all PRs contain test code.

**Comment by CITguy** [comment]
@glynnis: your feedback was not addressed with moving the section and renaming the heading.

File: src/elements/Tables/docs/tables.docs.html (hunk 8 above)
Comment (CITguy) [inline src/elements/Tables/docs/tables.docs.html]: @khaziwallis: please address this feedback.

File: src/elements/Tables/docs/tables.docs.html (hunk 7 above)
Comment (CITguy) [inline src/elements/Tables/docs/tables.docs.html]: @khaziwallis: please address this feedback.

**Comment by CITguy** [comment]
@khaziwallis, please rebase this PR

**Comment by satyamkotakonda** [comment]
@CITguy  Squash done, Please review.

File: src/elements/Tables/Tables.meta.json [hunk 9]
```diff
@@ -1,29 +1,35 @@
...
         "filter",
         "filtering",
         "floating header",
-        "nested",
+        "form",
+        "input",
         "nested content",
         "nested metadata",
         "nested table",
+        "nested",
```
Comment (100stacks) [inline src/elements/Tables/Tables.meta.json]: @satyamkotakonda, please move `nested` before `nested content`.

File: src/elements/Tables/docs/tables.docs.html [hunk 10]
```diff
@@ -186,6 +186,11 @@
 </ul>
 <rx-example name="table.filteringCollapsible"></rx-example>
 
+<h3><rx-permalink>Multi-select filters</rx-permalink></h3>
```
Comment (100stacks) [inline src/elements/Tables/docs/tables.docs.html]: @satyamkotakonda, capitalize title as `Multi-Select Filters`.

File: src/elements/Tables/docs/examples/rxSelectFilter.simple.html [hunk 11]
```diff
@@ -1,5 +1,5 @@
 <!-- Sample HTML goes here as a live example of how the component can be used -->
```
Comment (100stacks) [inline src/elements/Tables/docs/examples/rxSelectFilter.simple.html]: `rxSelectFilter.simple.html` needs to be updated:

* Per the HTML Styleguide, update contents to adhere to 2 spaces per indent.
* Remove comment at the top of the file.

**Comment by 100stacks** [comment]
@satyamkotakonda, please rename this file:

* `git mv src/elements/Tables/docs/examples/table.filteringCollapsible.less src/elements/Tables/docs/examples/table.filteringCollapsible.demo.less`

screen shot 2016-08-02 at 3 19 28 pm

This LESS file is for demo purposes only so we don't need to display it...thanks!

**Comment by satyamkotakonda** [comment]
@100stacks Updated changes and please review.
Thanks for your suggestions.

**Comment by 100stacks** [comment]
LGTM

**Comment by CITguy** [comment]
LGTM
